pkg/expression/generator/time_vec.go

```
@@ -299,30 +299,36 @@ func (b *{{.SigName}}) vecEval{{ .Output.TypeName }}(ctx EvalContext, input *chu
 			}
 		}
 	{{ else if or (eq .SigName "builtinAddDateAndDurationSig") (eq .SigName "builtinSubDateAndDurationSig") }}
-		fsp0 := b.args[0].GetType(ctx).GetDecimal()
 		fsp1 := b.args[1].GetType(ctx).GetDecimal()
 		arg1Duration := types.Duration{Duration: arg1, Fsp: fsp1}
+		tc := typeCtx(ctx)
+		arg0.SetType(mysql.TypeDatetime)
 		{{ if eq $.FuncName "AddTime" }}
-		sum, err := types.Duration{Duration: arg0, Fsp: fsp0}.Add(arg1Duration)
+		res, err := arg0.Add(tc, arg1Duration)
 		{{ else }}
-		sum, err := types.Duration{Duration: arg0, Fsp: fsp0}.Sub(arg1Duration)
+		res, err := arg0.Add(tc, arg1Duration.Neg())
 		{{ end }}
 		if err != nil {
-			return err
+			tc.AppendWarning(err)
+			result.AppendNull()
+			continue
```

why err is ignored?

---

> why err is ignored?

Because mysql ignores this err.